_posts/2021-10-06-jvm-issues-at-comcast.md

```diff
@@ -0,0 +1,219 @@
+---
+layout: post
+title:  "JVM Challenges in Production"
+author: Sajumon Joseph, David Leach, Bryan Aller, Pavan Madhineni, Lavanya Ragothaman, Pratap Moturi, Pádraig O'Sullivan (Starburst)
+excerpt_separator: <!--more-->
+---
+
+At Comcast, we have a large on-premise Trino cluster. It enables us to extract
+insights from data no matter where it resides and prepares the company for a
+more cloud-centric future. Recently, however, we experienced and overcame
+challenges related to the JVM. We wanted to share what we encountered and
+learned in hopes that it might be useful for the Trino community.
+
+<!--more-->
+
+## JIT Recompilations
+
+Some users complained that nightly reports were taking far too long to
+complete. Queries that ran for 6 hours made very little progress.
+
+First, we looked at the queries involved in these nightly reports. We
+noticed that all these queries involved 2 particular tables. In this post,
+we will call them table A and table B.
+
+Our initial suspicion was that there could be an issue with the table data in
+HDFS. Thus, we tried to reproduce the performance problem by using queries that
+performed simple scans against these tables. 
+
+We tried a simple table scan with no filters, range filter on a partitioned
+column, etc.,  ran these queries multiple times and execution times were
+consistent. This ruled out a potential problem with HDFS.
+
+Next, we took a closer look  at the portion of the slow running queries
+involving the A table and came up with the simplest possible query that could
+demonstrate the problem. We discovered that the following query performed as
+expected:
+
+```
+select
```

Comment:
For here and all statements .. use uppercase SQL keywords like in the docs

SELECT count(a.c) 
FROM
...